conda/exports.py

```diff
@@ -132,6 +135,12 @@ class memoized(object):  # pragma: no cover
     (not reevaluated).
     """
     def __init__(self, func):
+        warnings.warn(
+            "The `conda.exports.memoized` is pending deprecation and will be removed in a future "
+            "release. Please use `functools.lru_cache` instead.",
+            PendingDeprecationWarning,
```

Comment:
I'm wondering if it would be better to use a stacklevel of 2 here to make sure the deprecation warning shows up next to the code that uses the decorator than pointing to the implementation here.

```suggestion
            PendingDeprecationWarning, stacklevel=2
```

But OTOH I'm not sure how decorator initialization behaves in this case with the stack.

Reply:
I don't think the `stacklevel` helps here. This is what I get without the `stacklevel` which correctly refers to `conda.exports` line 138:

![](https://user-images.githubusercontent.com/4546435/177852886-07383cf8-4cff-4661-8ea4-1810342ecd5a.png)

And with `stacklevel` it refers to `stdin`:

![](https://user-images.githubusercontent.com/4546435/177853121-7e59ede4-f8a1-4e41-9ef9-920c7198c855.png)

Reply:
Ah, would you mind trying with a stacklevel of 1?

Reply:
Wait, I think you're right and I'm wrong. We do want stacklevel 2 because we want the second case from above (not the first), we want the `<stdin>:2` file/line. We need users to see where _they_ have included `@memoized` not where `memoized` is defined in conda, right?